File: Homeworks/HW1/Codes/Dogan_MuhammedIkbal.php
<?php
$numbers = array("Ahmet"=>"35", "Mehmet"=>"37", "Ali"=>"43");// 1 - initialize
echo $numbers["Ahmet"]; // 2 get a value for given key
echo "<br>";
print_r($numbers);
echo "<br>";
$numbers += ["Ayse" => "15"]; // 3- adding an element
print_r($numbers);
unset($numbers["Mehmet"]); // 4 - removing an element
echo "<br>";
print_r($numbers);
echo "<br>";
$numbers["Ali"] = "18"; // 5- modifying a value
print_r($numbers);
echo "<br>";
if (array_key_exists("Ali",$numbers)) // 6- checking existence of key
  {
  echo "Key exists!";
  }
else
  {
  echo "Key does not exist!";
  }
echo "<br>";
if (in_array("18",$numbers)) // 7- checking existence of value
  {
  echo "18 exists!";
  }
else
  {
  echo "18 not exist!<br>";
  }
function foo($mymap)// 8- printing key value pairs
{
	foreach($mymap as $key => $value)
	{
  		echo $key." has the value ". $value . "<br>";
	}

}
foo($numbers);
?>

File: Homeworks/HW1/Codes/Dogan_MuhammedIkbal.dart
void main() {

  Map<int,String> map = { 0:'murat'}; //1-initialize
  
  print(map[0]); // 2- get value for key
  map[1] = 'Ahmet';
  map[2] = 'Mehmet'; //3- adding new elements
  map[3] = 'Veli';
  print(map[1]);
  print (map);
  map.remove(1); //4-removing an element
  print (map);
  map[2] = ('Ayse'); // 5- modifying an element
  print (map);
  print(map.containsKey(1)); // 6- search for existence of a key
  print (map.containsValue('Ayse')); // 7- search existence of a value
  
  foo(map); // printing elements by function
}

void foo (Map<dynamic,dynamic> map){
  
  map.forEach((k,v) => print('${k}: ${v}'));
  
}